sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
+// type, but are sufficiently into the future/past.
+const (
+	hundredBillion = 100 * 1000 * 1000 * 1000
+)
+
+var maxTime = time.Date(hundredBillion, time.January, 1, 0, 0, 0, 0, time.UTC)
+var minTime = time.Date(-hundredBillion, time.January, 1, 0, 0, 0, 0, time.UTC)
+
+// DDate is the date Datum.
+type DDate struct {
```

Comment:
And the alternative was? If you declare `DDate` as `type DDate time.Time` then you have to cast to `time.Time` in order to access its methods. This seems cleaner.